[raw-asparagus] iP

raw-asparagus opened the pull request:
# Dusk: Your Personal Task Manager :DD

> "Did all the extra credit, then got graded on a curve. I think it's time to teach some lessons." – [Taylor A. Swift](https://genius.com/Taylor-swift-bejeweled-lyrics)

Dusk is here to aid you track all you to-dos! 

## Getting Started

1. download it from [here](#).
2. double-click the `Dusk.jar` file
3. chat with Dusk!
4. let it manage your tasks for you :))

And it is **FREE**!!1!!!1!

### Sample Usage

<To be added>

## Current Status

- [x] Managing tasks
- [x] Search tasks
- [ ] Sample usage
- [ ] 
- [ ] 

---

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

raw-asparagus committed:
refactor: refactor duke into dusk

raw-asparagus committed:
feat: add basic chat to dusk

raw-asparagus committed:
feat: add message history

raw-asparagus committed:
style: abstract tasks to Task class

raw-asparagus committed:
style: change IO error logging

raw-asparagus committed:
feat: add to-do checklist

raw-asparagus committed:
style: support inheritance for more task types

raw-asparagus committed:
feat: add new task types - todo, deadline, event

raw-asparagus committed:
feat: add more command flags

raw-asparagus committed:
feat: add more command flags

raw-asparagus committed:
dev: add automated text ui testing

raw-asparagus committed:
style: refactor to use more switch-case

raw-asparagus committed:
dev: add exception handling

raw-asparagus committed:
dev: use Java Collections

raw-asparagus committed:
feat: add delete feature

raw-asparagus committed:
style: refactor switch-case condition to use enums

raw-asparagus committed:
dev: update testcase

raw-asparagus committed:
dev: refactor command functions into Command classes

raw-asparagus committed:
feat: add FileNotFoundException.java to handle file not found errors

raw-asparagus committed:
feat: add FileNotFoundException.java to handle file not found errors

raw-asparagus committed:
dev: protected Task class variables rename

# Conflicts:
#	src/main/java/Task.java

raw-asparagus committed:
dev: update IO handling

raw-asparagus committed:
dev: split Dusk.java

raw-asparagus committed:
feat: add asynchronous storage

raw-asparagus committed:
Merge pull request #1 from raw-asparagus/branch-Level-7

branch-Level-7

raw-asparagus committed:
dev: split Dusk.java

raw-asparagus committed:
dev: split Dusk into clases

raw-asparagus committed:
dev: create storage, task and ui package

raw-asparagus committed:
dev: move task list operations to task package

raw-asparagus committed:
dev: minor syntax update

raw-asparagus committed:
dev: refactor Dusk

raw-asparagus committed:
refactor to reduce dependency between packages

raw-asparagus committed:
dev: refactor to reduce dependency between classes

raw-asparagus committed:
Merge pull request #2 from raw-asparagus/branch-A-MoreOOP

branch-A-MoreOOP

raw-asparagus committed:
dev: fix loading from data file

raw-asparagus committed:
dev: create command package

raw-asparagus committed:
dev: enforce private scoping of methods

raw-asparagus committed:
dev: move dusk into dusk package

raw-asparagus committed:
Merge pull request #3 from raw-asparagus/branch-A-Packages

branch-A-Packages

raw-asparagus committed:
dev: move other packages into dusk package

raw-asparagus committed:
dev: refactor dusk to remove cyclic dependency

raw-asparagus committed:
Merge pull request #4 from raw-asparagus/branch-A-MoreOOP

branch-A-MoreOOP

raw-asparagus committed:
feat: add task datetime formatting support

raw-asparagus committed:
dev: update testcases and code linebreak reformat

raw-asparagus committed:
dev: update testcases and code linebreak reformat

raw-asparagus committed:
dev: add list by date/range support

raw-asparagus committed:
dev: update test case

raw-asparagus committed:
feat: add exclamation mark

raw-asparagus committed:
Merge pull request #5 from raw-asparagus/branch-Level-8

branch-Level-8

raw-asparagus committed:
Merge pull request #6 from nus-cs2103-AY2425S2/add-gradle-support

add-gradle-support

raw-asparagus committed:
Merge pull request #7 from raw-asparagus/add-gradle-support

add-gradle-support

KimHan01 commented on src/main/java/dusk/command/Parser.java:
```diff
@@ -0,0 +1,113 @@
+                        case "to" -> toStr = foundValue;
+                        case "by" -> byStr = foundValue;
+                        default -> {
+                        }
```
Great job following the switch statement coding standard for Lambda-style!

KimHan01 commented on src/main/java/dusk/storage/Storage.java:
```diff
@@ -0,0 +1,175 @@
+
+    private Task getDataFile(String[] parts) throws StorageException {
+        String taskType = parts[0];
+        boolean done = Boolean.parseBoolean(parts[1]);
```
Boolean variables should be named to sound like booleans. Perhaps a name like "isDone" can be a good alternative!

KimHan01 commented on src/main/java/dusk/task/Task.java:
```diff
@@ -0,0 +1,35 @@
+
+    public boolean getDone() {
+        return isDone;
+    }
```
Should boolean methods also be named to sound like booleans?

KimHan01 reviewed (commented):
Apart from a few nits, you followed the coding standard really well!
Keep it up 👍

raw-asparagus committed:
dev: add JUnit testing for create command classes

raw-asparagus committed:
dev: add JUnit testing

raw-asparagus committed:
Merge pull request #8 from raw-asparagus/branch-A-JUnit

branch-A-JUnit

raw-asparagus committed:
dev: add jar export

raw-asparagus committed:
Merge pull request #9 from raw-asparagus/branch-A-Jar

branch-A-Jar

raw-asparagus committed:
dev: add JavaDoc for Dusk.java

raw-asparagus committed:
dev: add JavaDoc for ui package

raw-asparagus committed:
dev: add JavaDoc for task package

raw-asparagus committed:
dev: add JavaDoc for storage package

raw-asparagus committed:
dev: add JavaDoc for command package

raw-asparagus committed:
dev: add JavaDoc for test

raw-asparagus committed:
Merge pull request #10 from raw-asparagus/branch-A-JavaDoc

branch-A-JavaDoc

raw-asparagus committed:
dev: enforce coding standards for package command

raw-asparagus committed:
dev: enforce coding standards for package storage

raw-asparagus committed:
dev: enforce coding standards for package task

raw-asparagus committed:
dev: enforce coding standards for package io

raw-asparagus committed:
dev: enforce coding standards for Dusk.java

raw-asparagus committed:
dev: enforce coding standards for test

raw-asparagus committed:
Merge pull request #11 from raw-asparagus/branch-A-CodingStandard

branch-A-CodingStandard

arshinsikka commented on src/main/java/dusk/command/MarkCommand.java:
```diff
@@ -0,0 +1,67 @@
+    private final ConsoleIO consoleIO;
+    private final Storage storage;
+    private final String description;
+    private final boolean markAsDone;
```
Rename markAsDone → shouldMarkAsDone for clarity, more conventional for boolean fields naming

arshinsikka commented on src/main/java/dusk/command/MarkCommand.java:
```diff
@@ -0,0 +1,67 @@
+     * @param description the index of the task to mark/unmark
+     * @param markAsDone  whether to mark the task as done (true) or not (false)
+     */
+    public MarkCommand(TaskList tasks, ConsoleIO consoleIO, Storage storage,
```
The class MarkCommand handles both marking and unmarking tasks. A clearer name would be ToggleMarkCommand or UpdateTaskStatusCommand.
Suggestion: Rename MarkCommand → ToggleMarkCommand

arshinsikka commented on src/main/java/dusk/command/Parser.java:
```diff
@@ -0,0 +1,127 @@
+     * @return the command object corresponding to the user input
+     * @throws InputException if the command is invalid or incorrectly formatted
+     */
+    public static Command parse(ConsoleIO consoleIO, Storage storage,
```
The parse method in Parser is generic, but it specifically parses commands.
Suggestion: Rename parse → parseCommand.

arshinsikka reviewed (commented):
Overall, not many errors.

raw-asparagus committed:
feat: add find command

raw-asparagus committed:
Merge branch 'master' into branch-Level-9

raw-asparagus committed:
Merge pull request #12 from raw-asparagus/branch-Level-9

branch-Level-9

raw-asparagus committed:
Add support for "find" command in the parser

raw-asparagus committed:
Merge pull request #14 from raw-asparagus/branch-Level-9

Add support for "find" command in the parser

raw-asparagus committed:
Enable Checkstyle and refactor code for compliance

raw-asparagus committed:
Add assertInstanceOf import to ParserTest

raw-asparagus committed:
Merge pull request #15 from raw-asparagus/branch-A-CheckStyle

Branch A-CheckStyle

raw-asparagus committed:
Add GUI support and refactor input processing

raw-asparagus committed:
Refactor command handling and streamline ConsoleIO interactions

raw-asparagus committed:
Rename ConsoleIO to DuskIO for clarity and consistency

raw-asparagus committed:
Rename ConsoleIO to DuskIO for clarity and consistency

raw-asparagus committed:
Refactor build.gradle for better maintainability and clarity.

raw-asparagus committed:
Merge pull request #16 from raw-asparagus/branch-Level-10

branch-level-10

raw-asparagus committed:
Fix null pointer exception during window closure

raw-asparagus committed:
Refactor delayed termination logic for

raw-asparagus committed:
Revert "Refactor delayed termination logic for"

This reverts commit e7659350edb30287c89f4e240d50c063964e5833.

raw-asparagus committed:
Revert "Fix null pointer exception during window closure"

This reverts commit 1afc2f0f3c47989780b9f00785021eec14944e60.

raw-asparagus committed:
Fix potential null pointer exception during window close

raw-asparagus committed:
Refactor FXML files to remove unnecessary <children> tags

raw-asparagus committed:
Remove unused constants and deprecated method

raw-asparagus committed:
Merge pull request #17 from raw-asparagus/branch-a-varargs

branch-a-varargs

raw-asparagus committed:
Add null checks and assertions for key instance variables

Ensure critical objects like Dusk, dialogContainer, and response are non-null before use. Added assertions to improve debugging and prevent potential runtime issues caused by null inputs or uninitialized elements.

raw-asparagus committed:
Merge pull request #18 from raw-asparagus/branch-A-Assertions

Add null checks and assertions for key instance variablesa

raw-asparagus committed:
Refactor imports and improve code formatting consistency

Reorganize import statements across classes for better readability and maintainability. Adjust code formatting to ensure consistent alignment and spacing in parameters and comments. No functional changes were introduced.

raw-asparagus committed:
Merge branch 'branch-A-CodeQuality' into master

raw-asparagus committed:
Merge pull request #19 from raw-asparagus/master

branch-A-CodeQuality

raw-asparagus committed:
Merge pull request #20 from raw-asparagus/branch-A-CodeQuality

Merge pull request #19 from raw-asparagus/master

raw-asparagus committed:
Refactor task iteration to use streams forEach.

Replaced the for-loop with a streams forEach method to enhance readability and modernize the code. This change simplifies the iteration logic and aligns the code with functional programming practices.

raw-asparagus committed:
Merge pull request #21 from raw-asparagus/master

branch-A-Streams

raw-asparagus committed:
Merge pull request #22 from raw-asparagus/branch-A-Streams

branch-A-Streams

raw-asparagus committed:
Add GitHub Actions workflow for Java CI with Gradle

This workflow runs on push and pull requests, testing across Ubuntu, macOS, and Windows environments. It sets up JDK 17, validates the Gradle Wrapper, and performs a Gradle build and check step to ensure code quality.

raw-asparagus committed:
Merge pull request #23 from raw-asparagus/branch-A-CI

branch-A-CI

raw-asparagus committed:
Refactor exception handling to introduce DuskException hierarchy

raw-asparagus committed:
Add a logger to replace stack trace printing in Main.java

raw-asparagus committed:
Update logger class references and improve error handling

raw-asparagus committed:
Refactor Dusk responses to include type and metadata with better

raw-asparagus committed:
Refactor DialogBox to support error-specific styling

raw-asparagus committed:
Fix Java CI build issue from DuskException refactor

raw-asparagus committed:
Refactor: Move exception and response classes to dusk.ui

raw-asparagus committed:
Add custom styling to UI components and integrate stylesheet

raw-asparagus committed:
Merge pull request #24 from raw-asparagus/branch-A-BetterGui

Branch-A-BetterGui

raw-asparagus committed:
Update JavaFX version and enforce Java 17 compatibility

raw-asparagus committed:
Refactor task search and improve date handling

raw-asparagus committed:
Refactor and enhance command parsing logic

raw-asparagus committed:
Standardize error messages and improve async error handling

raw-asparagus committed:
Enhance FindCommand to support partial keyword matches

raw-asparagus committed:
Merge pull request #25 from raw-asparagus/branch-C-BetterSearch

branch-C-BetterSearch

raw-asparagus committed:
Set scroll pane policies and improve I/O validation and formatting

raw-asparagus committed:
Refactor dialog handling and improve FXML consistency

raw-asparagus committed:
Improve UI scrolling and refactor inline styles to CSS

raw-asparagus committed:
Refactor command classes and improve input validation

raw-asparagus committed:
Add Ui.png to the documentation folder

raw-asparagus committed:
Revamp README with detailed user guide and examples

raw-asparagus committed:
Update README.md with usage instructions and UI image

raw-asparagus committed:
Remove outdated README documentation for Dusk Task Manager

raw-asparagus committed:
Add user guide for Dusk Task Manager

raw-asparagus committed:
Merge pull request #26 from raw-asparagus/A-UserGuide

Branch-A-UserGuide

raw-asparagus committed:
Refactor UI classes for readability and documentation

raw-asparagus committed:
Refactor Storage.java and StorageException.java for clarity

raw-asparagus committed:
Refactor task and exception code for clarity and consistency

raw-asparagus committed:
Refactor exception classes by updating package structure

raw-asparagus committed:
Refactor codebase with improved documentation and clarity

raw-asparagus committed:
Add Javadoc for main method in Launcher class

raw-asparagus committed:
Add Mockito dependency for unit testing

raw-asparagus committed:
Remove unused import in ListCommand.java

raw-asparagus committed:
Refactor task creation commands into a shared abstract class

raw-asparagus committed:
Refactor boolean variable names for clarity

raw-asparagus committed:
Refactor from `getName` to `getDescription` and standardize response type enums

raw-asparagus committed:
Refactor and enhance StorageTest documentation and test cases

raw-asparagus committed:
Refactor StorageTest to use system-dependent file separator

raw-asparagus committed:
Remove invalidStorageLocationHandling test from StorageTest

raw-asparagus committed:
Merge pull request #27 from raw-asparagus/branch-A-AiAssisted

Update branch-A-MoreTesting

raw-asparagus committed:
Refactor and expand README documentation

raw-asparagus committed:
Refactor TaskListTest to improve structure and coverage

raw-asparagus committed:
Refactor test cases and update string representations

raw-asparagus committed:
Refactor `ParserTest` to simplify and improve test clarity

raw-asparagus committed:
Refactor and expand tests for MarkCommand; add DeleteCommandTest

raw-asparagus committed:
Refactor tests to use Mockito for mocking dependencies

raw-asparagus committed:
Refactor command test classes for clarity and organization

raw-asparagus committed:
Update AI.md declarations

raw-asparagus committed:
Merge pull request #28 from raw-asparagus/branch-A-MoreTesting

Improve JUnit and mockito testing

raw-asparagus committed:
Merge pull request #29 from raw-asparagus/branch-A-AiAssisted

Refactor and expand README documentation

raw-asparagus committed:
Refactor user input handling and add error message support

raw-asparagus committed:
Refactor dialog handling and replace ImageView with ProfilePicture

raw-asparagus committed:
Refactor GUI components and improve UI styling

raw-asparagus committed:
Merge pull request #30 from raw-asparagus/branch-A-BetterGui

Improve GUI and code quality
